src/main/java/org/candlepin/model/EnvironmentCurator.java

```diff
@@ -71,19 +69,88 @@ public List<Consumer> getEnvironmentConsumers(String environmentId) {
             .getResultList();
     }
 
-    public CandlepinQuery<Environment> listForOwner(Owner o) {
-        DetachedCriteria criteria = this.createSecureDetachedCriteria()
-            .add(Restrictions.eq("owner", o));
+    public List<Environment> listForOwnerByType(Owner o, List<String> type) {
+        String jpql = "SELECT e FROM Environment e WHERE e.owner.id = :owner_id and e.type in :type";
+        return this.getEntityManager()
+            .createQuery(jpql, Environment.class)
+            .setParameter("owner_id", o.getId())
+            .setParameter("type", type.stream().map(x -> x.toLowerCase()).toList())
+            .getResultList();
+    }
+
+    public List<Environment> listForOwnerIgnoreType(Owner o) {
```

Comment:
nit: naming convention for this type of query is generally `listAllBy[distinguishing criteria]`